interceptor.go

```diff
@@ -142,78 +63,66 @@ func (i *interceptor) WrapUnary(next connect.UnaryFunc) connect.UnaryFunc {
 				return next(ctx, request)
 			}
 		}
-		tracer := i.config.Tracer()
-		carrier := propagation.HeaderCarrier(request.Header())
-		attrs := attributesFromRequest(req)
+		isClient := request.Spec().IsClient
 		name := strings.TrimLeft(request.Spec().Procedure, "/")
-		protocol := protocolToSemConv(request.Peer())
-		var span trace.Span
-		if request.Spec().IsClient {
-			ctx, span = tracer.Start(
-				ctx,
-				name,
-				trace.WithSpanKind(trace.SpanKindClient),
-				trace.WithAttributes(attrs...),
```

Comment:
This basically meant that any span context passed into a client will not be propagated. I don't see any reason why clients shouldn't be able to pass in existing trace spans

Reply:
AFAIK this is pretty fundamental to tracing - new RPCs _always_ start new spans. This is certainly how the gRPC instrumentation works, and it's how tracing has worked in the handful of places I've seen it widely deployed.

Reply:
should there at least be a way to start traces in the clients? I can imagine a usecase where it might be useful to start the trace in the client so that all the separate rpc calls can be linked together (buf cli for example this might be useful)

Reply:
So the updated behaviour is:
- If client, then either use existing span from ctx, or create a new trace (default behaviour of tracer.Start)
    - Inject into carrier so that downstreams can use
- If Server then check if span in ctx; if so then use that one, otherwise use traceid from incoming http header.

All of this propagation can be turned on/off by using a custom `propagation.TextMapPropagator`